File: js-jasmine/src/conjured.js
class Conjured {

    constructor(item){
    this.item = item
  }
   decreaseSellin() {
    this.item.sellin -= 1;
  }
  decreaseQuality() {
    this.item.quality -= 2;
  }

  update() {
    this.decreaseSellin()

    if (this.item.sellIn < 0) {
      this.decreaseQuality()
    }
    if (this.item.quality > 0 && this.item.quality < 50) {
      this.decreaseQuality();
    }
    return this.item
  }
  }
module.exports = Conjured

File: Java/src/main/java/com/gildedrose/items/Conjured.java
package com.gildedrose;

public class Conjured {
    public Item update(Item conjured){
        if(conjured.sellIn > 0){
            conjured.quality = Math.max(0, conjured.quality -= 2);
        }else{
            conjured.quality = Math.max(0, conjured.quality -= 4);
        }
        conjured.sellIn--;
        return conjured;
    }
}
